### PR TITLE
RAC-467 Add possibility to have onCLick and on select on Card component

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: akeneo-design-system/src/components/Card/Card.stories.mdx [hunk 1]
```diff
@@ -148,8 +156,10 @@ Here the `fit` prop is also set to `contain`.
             isSelected={secondSelected}
             onSelect={isSelected => setSecondSelected(isSelected)}
           >
-            <Badge level="warning">75%</Badge>
-            Badge &amp; selectable 🤩
+            <Card.BadgeContainer>
+              <Badge level="warning">75%</Badge>
+            </Card.BadgeContainer>
+            <Card.Label>Badge &amp; selectable 🤩</Card.Label>
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Card/Card.stories.mdx:162]:
```suggestion
            Badge &amp; selectable 🤩
```

File: akeneo-design-system/src/components/Card/Card.tsx [hunk 2]
```diff
@@ -117,39 +128,46 @@ type CardProps = Override<
...
+  const nonLabelChildren: ReactElement[] = [];
+  const texts: string[] = [];
+
+  React.Children.forEach(children, child => {
+    if (typeof child === 'string') {
+      texts.push(child);
+    } else if (isValidElement(child)) {
+      nonLabelChildren.push(child);
+    } else {
+      throw new Error('Card component only accepts string or Badge as children');
```
Comment (StevenVAIDIE) [inline akeneo-design-system/src/components/Card/Card.tsx:150]:
```suggestion
      throw new Error('Card component only accepts string or Card.BadgeContainer as children');
```

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/legacy-bridge/src/provider/DependenciesProvider.type.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ type Router = {
 
 type Security = {isGranted: (acl: string) => boolean};
 
-type Translate = (id: string, placeholders?: {[name: string]: string}, count?: number) => string;
+type Translate = (id: string, placeholders?: {[name: string]: string|number}, count?: number) => string;
```
Comment (StevenVAIDIE) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/legacy-bridge/src/provider/DependenciesProvider.type.ts:21]:
```suggestion
type Translate = (id: string, placeholders?: {[name: string]: string | number}, count?: number) => string;
```

**Comment by StevenVAIDIE** [comment]
Mob programming

**Review by ValentinMumble — APPROVED** [review]

**Pull request merged**